### PR TITLE
IA-3051 allow building custom image from scripts directly without auto detecting versions or explicitly entering versions

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/IA-3051

https://fc-jenkins.dsp-techops.broadinstitute.org/job/leonardo-custom-image-rtitle/35/ is an example run (I aborted the actual run since I don't need the images, but the job looks good)

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: jenkins/Jenkinsfile [hunk 1]
```diff
@@ -151,12 +151,14 @@ pipeline {
...
             description: "The path for the bucket where the custom GCE image generation logs will be stored")
-        string(name: "DATAPROC_VERSIONS", defaultValue: "1.4.51-debian10",
+        string(name: "DATAPROC_VERSIONS", defaultValue: "2.0.24-debian10",
             description: "A custom image will be build for each of these dataproc versions")
         string(name: "branch", defaultValue: "develop",
             description: "The branch that will be used to run the job")
         booleanParam(name: "USE_CUSTOM_DOCKER_IMAGE_VERSIONS", defaultValue: false,
             description: "Check only if you want to manually specify all of the custom image versions below.")
+        booleanParam(name: "USE_DEFAULT_DOCKER_IMAGE_VERSIONS", defaultValue: false,
+            description: "When true, use docker image versions in scripts; when false, try to auto detects docker image versions, or use custom docker image versions from job input")
```
Comment (kyuksel) [inline jenkins/Jenkinsfile]: Nit:
```suggestion
            description: "When true, use docker image versions in 'prepare*image' scripts; when false, try to auto-detect docker image versions, or use custom docker image versions from job input")
```

**Review by xuscrbw — APPROVED** [review]
LGTM

**Review by kyuksel — APPROVED** [review]
LGTM with a nit inline

**Comment by Qi77Qi** [comment]
jenkins retest

**Comment by Qi77Qi** [comment]
jenkins retest

**Comment by Qi77Qi** [comment]
leonardo-build failed for some reason...jenkins retest

**Pull request merged**